Src/Support/Google.Apis.Auth/OAuth2/ServiceCredential.cs

```diff
@@ -173,9 +173,16 @@ public ServiceCredential(Initializer initializer)
         /// <inheritdoc/>
         public async Task InterceptAsync(HttpRequestMessage request, CancellationToken cancellationToken)
         {
-            var accessToken = await GetAccessTokenWithHeadersForRequestAsync(request.RequestUri.AbsoluteUri, cancellationToken).ConfigureAwait(false);
+            var accessToken = await GetAccessTokenWithHeadersForRequestAsync(GetAudience(request.RequestUri), cancellationToken).ConfigureAwait(false);
             AccessMethod.Intercept(request, accessToken.AccessToken);
             accessToken.AddHeaders(request);
+
+            static string GetAudience(Uri uri) =>
```

Comment:
Any reason for having a method for this rather than just an extra first line?

```csharp
var audience = new UriBuilder(request.RequestUri) { Query = "", Path = "" }.Uri.ToString();
var accessToken = await GetAccessTokenWithHeadersForRequestAsync(audience, cancellationToken).ConfigureAwait(false);
```

Reply:
Moving to the first line, I toyed with several other ways of getting the audience and the others were uglier.